pinot-spi/src/main/java/org/apache/pinot/spi/stream/StreamMetadataProvider.java

```diff
@@ -106,6 +107,33 @@ default Map<String, PartitionLagState> getCurrentPartitionLagState(
     return result;
   }
 
+  /**
+   * Fetches the list of available topics/streams
+   *
+   * @param timeout Timeout for fetching the list of topics. If this is null, the implementation should use a default
+   *                timeout.
+   * @return List of topics
+   */
+  default List<TopicMetadata> listTopics(Duration timeout) {
```

Comment:
I feel this `Duration timeout` is very specific to Kafka?

Reply:
either use empty argument for default behavior or put a map to allow customization

Reply:
yes. Other APIs don't support it. However, I think starting with this API might be helpful because, I have seen large timeouts in these API calls which results in a bad UX downstream.

Do you recommend having a simple `listTopics()` API. I can update to that instead?

Reply:
reverting to no arg API instead.